### PR TITLE
Project: Guess Who?

### PR DESCRIPTION
**Pull request by TinaJuren**
https://trusting-pare-82b4ac.netlify.app

During this project I found it challenging to follow the project's instructions and to find relevant and comprehensive information to solve the tasks. Within the team we had good discussions around various solutions, but still I stumbled in the dark when trying to implement the suggestions myself. My Javascript file is like a quilt with patches from various sources. 

If I had more time I would have:
- spent more time on figuring out why some of the code works and some don't.
- made the "Find out" button work for the default brown hair color. Currently it's not working and I have to choose another option before going back, choose brown hair color, and press "Find out"- then it works.
- Added more attributes to the characters.
- Made the restart button work.

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
         `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+    } else if (category === 'other') {
```
Comment (MarySparrow) [inline code/script.js:327]: indentation <--

File: code/script.js [hunk 2]
```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
...
+        alert(
+          `Yes, the person has ${value} eyes! Keep all that has ${value} eyes`
+       )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+        alert(
+          `No, the person doesn't have ${value} eyes! Remove all that have ${value} eyes`
+      )
     }
   }
```
Comment (MarySparrow) [inline code/script.js:358]: indentation for the above if statements, make sure the curly bracket belonging to each section is in the same _vertical_ line as statement, like this: 
```
if () {
} else if (){
  } else {
    }
```

File: code/script.js [hunk 3]
```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
...
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
-
+  if (suspect === secret.name) {
+    // 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseText.innerHTML= `Congratulations, you're guess on ${suspect} is correct!`
+  } else {
+    winOrLoseText.innerHTML= `Sorry, ${suspect} is not the right person!`
+  }
```
Comment (MarySparrow) [inline code/script.js:392]: indentation -->

File: code/script.js [hunk 4]
```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
...
+    board.style.display = "none"
+  }
+  
 // Invokes the start function when website is loaded
 start()
+console.log('Start function invoked: ', charactersInPlay)
+
+// Invokes the generateBoard function when website is loaded
+generateBoard()
+
```
Comment (MarySparrow) [inline code/script.js:405]: The generateBoard() can be removed as the invoke of start()  already has the generateBoard function inside of it.

File: code/script.js [hunk 5]
```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
...
+// Invokes the setSecret function when website is loaded
+setSecret()
+console.log('This is the secret person: ', secret)
 
 // All the event listeners
 restartButton.addEventListener('click', start)
+questions.addEventListener('change',() => { selectQuestion(questions.value)})
+filter.addEventListener('click', checkQuestion)
+playAgain.addEventListener('click', start)
```
Comment (MarySparrow) [inline code/script.js:414]: I wish to add a comment how to get the playAgain button to work, but it looks correct with eventListener and getElementById so I do not know how to advise here.

**Review by MarySparrow — CHANGES_REQUESTED** [review]
Great job Tina! The code is very easy to read and follow the flow of the game! You did it! (insert fireworks!)

File: code/index.html [hunk 6]
```diff
@@ -26,14 +26,24 @@ <h1>Does the person have</h1>
         <optgroup label="hair color">
           <option value="brown">brown hair</option>
```
Comment (KarolCislowski) [inline code/index.html:27]: `<option value="brown" selected>brown hair</option>` should fix your problem with the brown hairs 🤓

File: code/script.js [hunk 7]
```diff
@@ -254,46 +259,62 @@ const start = () => {
...
+      value: value,
       category: category,
-    }
+    }                 
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category,
```
Comment (KarolCislowski) [inline code/script.js:283]: when your key and name of the variable used as value are the same as here you can use `{value, category}`

File: code/script.js (hunk 5 above)
Comment (KarolCislowski) [reply code/script.js:414]: You need to add 2 thinks to the start function- hide the win/lose page and show the board 🤓

**Review by KarolCislowski — APPROVED** [review]
This is a very nice project 🌈 Hope my tips will solve some of your issues 🦄 

I know it was an extra point of this assignment to style it your way but remember that there is no another way to master CSS then practice it over and over again 🦾 

Remember that you can always come back to it when you have some spare time and add more functionalities to make it an even greater portfolio asset! 🦄

**Pull request closed without merging**